Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+TypeSignatureDisambiguation.swift

```diff
@@ -0,0 +1,437 @@
+    mutating func update(with member: Int) -> Int? {
+        let (inserted, _) = insert(member)
+        return inserted ? nil : member
+    }
+    
+    @inlinable
+    func contains(_ member: Int) -> Bool {
+        storage & Self.mask(member) != 0
+    }
+    
+    @inlinable
+    var count: Int {
+        storage.nonzeroBitCount
+    }
+    
+    @inlinable
+    func isSuperset(of other: Self) -> Bool {
+        // Provide a custom implementation since this is called frequently in `combinationsToCheck()`
+        (storage & other.storage) == other.storage
+    }
+    
+    @inlinable
+    func union(_ other: Self) -> Self {
+        .init(storage: storage | other.storage)
+    }
+    
+    @inlinable
+    func intersection(_ other: Self) -> Self {
+        .init(storage: storage & other.storage)
+    }
+    
+    @inlinable
+    func symmetricDifference(_ other: Self) -> Self {
+        .init(storage: storage ^ other.storage)
+    }
+    
+    @inlinable
+    mutating func formUnion(_ other: Self) {
+        storage |= other.storage
+    }
+    
+    @inlinable
+    mutating func formIntersection(_ other: Self) {
+        storage &= other.storage
+    }
+    
+    @inlinable
+    mutating func formSymmetricDifference(_ other: Self) {
+        storage ^= other.storage
+    }
+}
+
+extension _TinySmallValueIntSet: Sequence {
+    func makeIterator() -> Iterator {
+        Iterator(set: self)
+    }
+    
+    struct Iterator: IteratorProtocol {
+        typealias Element = Int
+        
+        private var storage: UInt64
+        private var current: Int = -1
+        
+        @inlinable
+        init(set: _TinySmallValueIntSet) {
+            self.storage = set.storage
+        }
+        
+        @inlinable
+        mutating func next() -> Int? {
+            guard storage != 0 else {
+                return nil
+            }
+            // If the set is somewhat sparse, we can find the next element faster by shifting to the next value.
+            // This saves needing to do `contains()` checks for all the numbers since the previous element.
+            let amountToShift = storage.trailingZeroBitCount + 1
+            storage >>= amountToShift
+            
+            current += amountToShift
+            return current
+        }
+    }
+}
+
+// MARK: Table
+
+/// A fixed-size grid of elements.
+private struct Table<Element> {
+    typealias Size = (width: Int, height: Int)
+    @usableFromInline
+    let size: Size
+    private let storage: ContiguousArray<Element>
+
+    @inlinable
+    init(width: Int, height: Int, initializingWith initializer: (_ buffer: inout UnsafeMutableTableBufferPointer) throws -> Void) rethrows {
+        size = (width, height)
+        let capacity = width * height
+        storage = try .init(unsafeUninitializedCapacity: capacity) { buffer, initializedCount in
+            var wrappedBuffer = UnsafeMutableTableBufferPointer(width: width, wrapping: buffer)
+            try initializer(&wrappedBuffer)
```

Comment:
I'm surprised that Swift allows you to call a closure like this, while executing an initializer, passing in a pointer to the storage.